ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -860,76 +952,37 @@ namespace OrdinalAxis {
          *
          * @return {number}
          */
-        axisProto.lin2val = function (
-            val: number,
-            fromIndex?: boolean
-        ): number {
+        axisProto.lin2val = function (val: number): number {
             let axis = this,
                 ordinal = axis.ordinal,
-                ordinalPositions = ordinal.positions,
-                ret;
+                ordinalPositions = ordinal.positions, // for the current visible range
+                ret = 0,
+                extendedOrdinalPositions = this.extendedOrdinalPositions;
 
-            // the visible range contains only equally spaced values
+            // The visible range contains only equally spaced values.
             if (!ordinalPositions) {
-                ret = val;
+                return val;
+            }
 
-            } else {
+            // If the value is not inside the plot area,
+            // use the extended positions.
+            // (array contains also points that are currently out of range).
 
-                let ordinalSlope = ordinal.slope,
-                    ordinalOffset = ordinal.offset,
-                    i = ordinalPositions.length - 1,
-                    linearEquivalentLeft,
-                    linearEquivalentRight,
-                    distance;
-
-
-                // Handle the case where we translate from the index directly,
-                // used only when panning an ordinal axis
-                if (fromIndex) {
-
-                    if (val < 0) { // out of range, in effect panning to the left
-                        val = ordinalPositions[0];
-                    } else if (val > i) { // out of range, panning to the right
-                        val = ordinalPositions[i];
-                    } else { // split it up
-                        i = Math.floor(val);
-                        distance = val - i; // the decimal
-                    }
+            // When iterating for the first time,
+            // get the extended ordinal positional and assign them.
+            if (axis.ordinal && !extendedOrdinalPositions) {
+                extendedOrdinalPositions = axis.ordinal.getExtendedPositions();
+                this.extendedOrdinalPositions = extendedOrdinalPositions;
+            }
 
-                // Loop down along the ordinal positions. When the linear
-                // equivalent of i matches an ordinal position, interpolate
-                // between the left and right values.
-                } else {
-                    while (i--) {
```

Comment:
Now we are not handling the case if you want to get the pixel of a value, that is not included in extendedOrdinalPosition.

Reply:
Indeed, the current implementation only works for points that are included in the `extendedOrdinalPositions` array. 
If not, we get the first point from that array.
But I'm not sure if the fix should also resolve that? 
What is the potential use case for that? When we might need that?

Reply:
We might not need that internally, but we will break `xAxis.toValue()`, which is widely used by developers. Compare results in  these demos:
- https://jsfiddle.net/BlackLabel/rh51sbw8/ (current)
- https://jsfiddle.net/BlackLabel/rh51sbw8/1/ (candidate)